hooks/checkout

```diff
@@ -36,9 +49,13 @@ copy_checkout_from_s3() {
 checkout() {
   git reset --hard
   git clean -ffxdq
-  git fetch -v origin HEAD
+
+  timeout --preserve-status "${GIT_REMOTE_TIMEOUT}" git fetch -v origin HEAD \
+    || exit "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
   git config remote.origin.fetch
-  git fetch -v --prune origin "${BUILDKITE_BRANCH}"
+  timeout --preserve-status "${GIT_REMOTE_TIMEOUT}" git fetch -v --prune origin "${BUILDKITE_BRANCH}" \
```

Comment:
I think both the command should not use `--preserve-status`, and only cause the plugin to exit with `GIT_REMOTE_TIMEOUT_EXIT_CODE` if it actually timed out, and not if it failed. You can do that by checking for the exit code 124.

Reply:
Good point 👍